### DISCUSSION TITLE
Modernize CMake

### DISCUSSION THREAD

**Discussion by alexreinking** [cnjinhao/nana#632] [Ideas]
I think it would be nice to fix up the CMake build here. It currently doesn't produce CMake-usable installs (and indeed prints a fairly extreme warning not to do this perfectly-standard thing). I'm a maintainer for [Halide's](https://github.com/halide/Halide) build system and would be happy to be a resource on this.

**Comment by qPCR4vir** [comment]
Hi!
The fix is: add to your call to cmake `-DNANA_CMAKE_INSTALL = ON`

I know this adds complexity to the job of a few large library repository maintainers, but simplifies the lives of end users, many of whom are not as good with cmake as repository maintainers.

**Reply by alexreinking** [reply]
With that enabled, the resulting packages are completely unusable, unfortunately. They don't export targets correctly.

**Reply by qPCR4vir** [reply]
> ...the resulting packages are completely unusable, unfortunately. They don't export targets correctly.

What are the errors?

I afraid this is why we don't support installing...
Is your cmake file installing/consuming nana in your public repository (link please)?

The:
```
message("The compiled Nana library will be installed in ${CMAKE_INSTALL_PREFIX}/lib")
message("The Nana include files will be installed in ${CMAKE_INSTALL_PREFIX}/include")
```

are telling you what you expected?

Normally it is simpler to just compile nana in the same cmake call, with your project. Your build system will take care of compiling only what your really need (for nana mostly, only the first time)
